src/vma/dev/ring_simple.cpp

```diff
@@ -391,9 +391,13 @@ bool ring_simple::attach_flow(flow_tuple& flow_spec_5t, pkt_rcvr_sink *sink)
 			ring_logdbg("MC FlowTag ID=%d is enabled as force_flowtag=%d SO_REUSEADDR=%d",
 				    flow_tag_id, m_b_sysvar_mc_force_flowtag, si->addr_in_reuse());
 		} else {
+			if (m_flow_tag_enabled) {
```

Comment:
It can be logically shown that `m_flow_tag_enabled == true` **if and only if** `flow_tag_id > 0`. This is due to the logic on lines 329 - 347 at the beginning of the function.
Therefor, the entire block (lines 390 - 400) can be greatly simplified by:

```
if (flow_tag_id) {
	if (m_b_sysvar_mc_force_flowtag || !si->addr_in_reuse()) {
		ring_logdbg("MC FlowTag ID ... is enabled ...");
	} else {
		flow_tag_id = FLOW_TAG_MASK;
		ring_logdbg("MC FlowTag ID ... is disabled ...");
	}
}
```

For the same reason, `m_flow_tag_enabled` can be removed from the test further down on line 495.